server/controllers/upload-controller.js

```diff
@@ -0,0 +1,312 @@
+const knex = require("./../db");
+const fs = require("fs");
+const csv = require("fast-csv");
+const moment = require("moment");
+const bcrypt = require("bcrypt");
+
+const clearFile = async (path) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      fs.unlinkSync(path);
+    } catch {
+      if (fs.existsSync(path)) {
+        clearFile(path);
+      }
+    }
+    if (!fs.existsSync(path)) resolve();
+    else reject();
+  });
+};
+
+const checkSupportOfficer = async (userId) => {
+  const user = await knex
+    .select({ role: "role" })
+    .from("user")
+    .where("id", userId)
+    .catch(() => {
+      throw { msg: `There was an error retrieving user info`, status: 400 };
+    });
+  if (user.length == 1 && !["supportOfficer"].includes(user[0].role)) {
```

Comment:
When there's just one string to compare it's better to use the standard comparison.
Also, remember to use strict comparison: `===` instead of `==`.
```suggestion
  if (user.length === 1 && "supportOfficer" !== user[0].role) {
```